### PR TITLE
Add cluster consistency parsing to cluster package

### PR DESCRIPTION
**Pull request by otoolep**
Right now failed parsing is indicated by returning a nil Graphite object. This will surely blow up later, but I don't any other component checking for failed services either.

Any suggestions @benbjohnson ?

### CODE REVIEW

**Comment by otoolep** [comment]
Alternatively we can just merge this, and move onto dealing with potential nil services another day.

**Comment by toddboom** [comment]
+1

File: cluster/points_writer.go [hunk 1]
```diff
@@ -39,8 +40,27 @@ var (
 
 	// ErrWriteFailed is returned when no writes succeeded.
 	ErrWriteFailed = errors.New("write failed")
+
+	// ErrConsistencyLevelParseError is returned when parsing the string version
+	// of a consistency level.
+	ErrConsistencyLevelParseError = errors.New("write failed")
```
Comment (benbjohnson) [inline cluster/points_writer.go]: Can you rename to `ErrInvalidConsistencyLevel`? Also, the error message doesn't match.

File: services/graphite/service.go [hunk 2]
```diff
@@ -46,16 +46,21 @@ type Service struct {
...
+		batchTimeout: time.Duration(c.BatchTimeout),
+		logger:       log.New(os.Stderr, "[graphite] ", log.LstdFlags),
+		done:         make(chan struct{}),
 	}
 
+	consistencyLevel, err := cluster.ParseConsistencyLevel(c.ConsistencyLevel)
+	if err != nil {
+		return nil
+	}
+	s.consistencyLevel = consistencyLevel
```
Comment (benbjohnson) [inline services/graphite/service.go]: I think you should return an error from `NewService()`:

```go
func NewService(c Config) (*Service, error)
```

File: cluster/points_writer.go (hunk 1 above)
Comment (otoolep) [inline cluster/points_writer.go]: Grrr, on error message not matching.

**Comment by otoolep** [comment]
Feedback from @benbjohnson addressed. I think adding `error` support to the `NewServer` calls is fine, but does ripple through the code. I think this is good, as server-creation can be complex, and there is no harm checking if it succeeded.

**Comment by benbjohnson** [comment]
:+1:

**Pull request merged**